src/libexpr/parser-state.hh

```diff
@@ -291,12 +291,22 @@ inline Expr * ParserState::stripIndentation(const PosIdx pos,
                 s2 = std::string(s2, 0, p + 1);
         }
 
-        es2->emplace_back(i->first, new ExprString(std::move(s2)));
+        // Ignore empty strings for a minor optimisation and AST simplification
+        if (s2 != "") {
+            es2->emplace_back(i->first, new ExprString(std::move(s2)));
+        }
     };
     for (; i != es.end(); ++i, --n) {
         std::visit(overloaded { trimExpr, trimString }, i->second);
     }
 
+    /* If there is nothing at all, return the empty string directly */
+    if (es2->size() == 0) {
+        auto *const result = new ExprString("");
+        delete es2;
+        return result;
+    }
+
```

Comment:
The problem I discovered is fixed by this. This is necessary because newly `es2` can be entirely empty, in which case we should not create an `ExprConcatStrings`, but just return an empty string directly. Not doing so would give inconsistent parsing output from
```nix
''''
```

and
```nix
''
''
```

which is now covered in the test.